src/core/Fingerprint.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "hyprlock.hpp"
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <condition_variable>
+#include <sdbus-c++/sdbus-c++.h>
+
+class CFingerprint {
+  public:
+    CFingerprint();
+
+    std::shared_ptr<sdbus::IConnection> start();
+    bool                                isAuthenticated();
+    std::optional<std::string>          getLastMessage();
+    void                                terminate();
+
+  private:
+    enum class MatchResult {
```

Comment:
C++: `enum`
Also, please:
- call enum variables in CAPS and with an identifiable prefix (e.g. `MATCH_NO_MATCH`, `MATCH_RETRY`)
- add a variable for invalid e.g. `MATCH_INVALID = 0`
- have the _first_ enum value be defined (e.g. as above = 0)

Reply:
Done